### PR TITLE
Advertising Function Cleanup

### PR DESCRIPTION
**Pull request by JaydenIrwin**
Cleaned up and improved performance of marketing_get_campaign_guest_generation_probability() by:

- Removed static constant array that was only used in 1 function
- Only defined "ride" within necessary scope
- Switch cases clearly show the relationship between enum case and it's value

### CODE REVIEW

**Comment by Gymnasiast** [comment]
Could you clean up your commit history? Do so by rebasing and squashing.

File: src/openrct2/management/Marketing.cpp [hunk 1]
```diff
@@ -36,35 +36,42 @@ const money16 AdvertisingCampaignPricePerWeek[] =
...
-    case ADVERTISING_CAMPAIGN_PARK_ENTRY_HALF_PRICE:
-        if (park_get_entrance_fee() < 6)
-            probability /= 8;
-        break;
-    case ADVERTISING_CAMPAIGN_RIDE_FREE:
-        ride = get_ride(gMarketingCampaignRideIndex[campaign]);
-        if (ride->price < 3)
-            probability /= 8;
-        break;
+        case ADVERTISING_CAMPAIGN_PARK_ENTRY_FREE:
```
Comment (Gymnasiast) [inline src/openrct2/management/Marketing.cpp]: Incorrect indentation. These are labels.
Indentation is as follows:
```
switch (var)
{
case 0:
    doStuff();
```

File: src/openrct2/management/Marketing.cpp [hunk 2]
```diff
@@ -36,35 +36,42 @@ const money16 AdvertisingCampaignPricePerWeek[] =
...
+            probability = 400;
+            if (park_get_entrance_fee() < 4)
+                probability /= 8; // Lower probability of guest generation if price was already low
```
Comment (Gymnasiast) [inline src/openrct2/management/Marketing.cpp]: Do not put comments next to code. Put it above the if.

**Comment by Gymnasiast** [comment]
After looking through, I have to say I don't think this particularly improves the code, to be brutally honest. The relationship between guest generation probabilities is a point, but that would be easily solved by having the guest generation table written vertically, with the type of the campaign in comments next to it.

I'm slightly reluctant to just close this, however, since I don't want to break your spirit, seeing as this is your first comment, but I'm afraid this isn't really up to snuff.

**Comment by JaydenIrwin** [comment]
Yes, this is my first time. So what does this mean? Does it mean I should make the requested changes and submit another PR? Or should I give up on this?

**Comment by Gymnasiast** [comment]
Hm, it's hard to say. I'm not convinced that moving the probabilities into the function is going to improve stuff, even though I understand why you did it.

What you _can_ do: 
```
static const sint32 AdvertisingCampaignGuestGenerationProbabilities[] = 
{
    400, // ADVERTISING_CAMPAIGN_x
```

For the above, open a new PR and make sure you have a clean commit history. I can help you with that if you need.

**Comment by JaydenIrwin** [comment]
Narrowing scope is always better for performance. And do you think making `ride` a variable was good, or do you think skipping that so I don’t need those brackets would be nicer looking?

I’m on Xcode, but it looks like I’ll need to use commands to clean the history. Yes, help would be great.

**Comment by Gymnasiast** [comment]
> Narrowing scope is always better for performance. 

That is quite a generalisation if you ask me.

Any, to rebase on macOS:

Open a Terminal and navigate to the directory where the OpenRCT2 code resides.
Now do (assuming you have our repository as `upstream`):
```
git checkout marketing_cleanup
git fetch upstream
git rebase -i upstream/develop
```
This will open Vim. Press `i` to put it into insert mode. Now change the word `pick` on the first line to `reword`, and change the word `pick` to `f` on other lines. Press <ESC> : x <ENTER> to quit and save.

It will now rebase. At some point it will ask for a new commit message (since you entered 'reword' for the first commit). Change the commit message to something more descriptive.

If it complains about conflicts, solve them manually, then do
```
git add -A
git rebase --continue
```

When you're, do
```
git push -f
```
The force parameter is required, since you're rewriting the history. Before you force push, check if everything is as it should be.

**Pull request closed without merging**